tvb_framework/tvb/tests/framework/adapters/uploaders/bids_zip_importer_test.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+#
+#
+# TheVirtualBrain-Framework Package. This package holds all Data Management, and
+# Web-UI helpful to run brain-simulations. To use it, you also need do download
+# TheVirtualBrain-Scientific Package (for simulators). See content of the
+# documentation-folder for more details. See also http://www.thevirtualbrain.org
+#
+# (c) 2012-2022, Baycrest Centre for Geriatric Care ("Baycrest") and others
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with this
+# program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+#   CITATION:
+# When using The Virtual Brain for scientific publications, please cite it as follows:
+#
+#   Paula Sanz Leon, Stuart A. Knock, M. Marmaduke Woodman, Lia Domide,
+#   Jochen Mersmann, Anthony R. McIntosh, Viktor Jirsa (2013)
+#       The Virtual Brain: a simulator of primate brain network dynamics.
+#   Frontiers in Neuroinformatics (7:10. doi: 10.3389/fninf.2013.00010)
+#
+#
+
+import os
+import pytest
+from tvb.adapters.uploaders.bids_importer import BIDSImporterModel, BIDSImporter
+from tvb.adapters.datatypes.db.connectivity import ConnectivityIndex
+from tvb.adapters.datatypes.db.surface import SurfaceIndex
+from tvb.adapters.datatypes.db.time_series import TimeSeriesIndex
+from tvb.adapters.datatypes.db.graph import CorrelationCoefficientsIndex
+from tvb.core.entities.transient.structure_entities import DataTypeMetaData
+from tvb.tests.framework.core.base_testcase import BaseTestCase
+from tvb.tests.framework.core.factory import TestFactory
+
+try:
+    import tvb_data.bids
+
+    BIDS_DATA_FOUND = True
+except ImportError:
+    BIDS_DATA_FOUND = False
+
+
+@pytest.mark.skipif(not BIDS_DATA_FOUND, reason="Older or incomplete tvb_data")
```

Comment:
@popaula937 @davidbacter01 Do you think this is reasonable for the time being, as we can not have a buds demo dataset published yet on Zenodo ?

Reply:
I agree with this change. Until the dataset is published, we should run this test locally whenever changes are made to the BIDS importer.

Reply:
Also, in the current form of the code, even after merge, the new BIDS derivatives Importer is not visible in TVB web GUI, until Charite gives us an OK on it. The code being merged into master is not much different than having it in a slowly deprecating branch, but at least we don't need to rebase and keep the branch in synch.

Reply:
Fully agree to merge it even if is not visible, would be easier to maintain.